h/views/accounts.py

```diff
@@ -269,9 +273,15 @@ def post(self):
                 self.form.set_widgets({"user": deform.widget.HiddenWidget()})
             return {"form": self.form.render()}
 
-        self._reset_password(appstruct["user"], appstruct["password"])
+        user = appstruct["user"]
+
+        self._reset_password(user, appstruct["password"])
 
-        return httpexceptions.HTTPFound(location=self.request.route_path("index"))
+        return httpexceptions.HTTPFound(
+            location=self.request.route_path(
+                "login", _query={"username": user.username}
+            )
+        )
```

Comment:
This has changed to redirect to login not index after resetting your password, and to add the username query param so that the username form field gets pre-filled